### PR TITLE
[Model Deactivation] Comply in all actions to whitelisted providers

### LINKED ISSUE #5396: [Model Deactivation] - Restrictions on `DustProdActionRegistry` interactions
We need to adapt title generation, instructions suggestions, etc... based on the activated/deactivated providers

- [ ] Make sure all interactions work with each provider we support
- [ ] Adapt model's used for those interactions to workspace's restrictions

### PR DESCRIPTION
**Pull request by spolu**
## Description

Fixes https://github.com/dust-tt/dust/issues/5396

- Remove default models from actions configs
- Dry out `isProviderWhitelisted`
- Introduce `getSmallWhitelistedModel` and `getLargeWhitelistedModel` for actions that are outside of the conversation (for which we already have guarantee that the model is whitelisted)

## Risk

High, all actions where tested in dev. Document tracker is no risk since only available to us

## Deploy Plan

- deploy `front`

### CODE REVIEW

**Comment by spolu** [comment]
cc @fontanierh for document tracker refactor to use `owner` in place of `workspaceId`

File: types/src/front/lib/assistant.ts [hunk 1]
```diff
@@ -50,6 +51,50 @@ export function metaPromptForProvider(
...
+  }
+  if (isProviderWhitelisted(owner, "mistral")) {
+    return MISTRAL_SMALL_MODEL_CONFIG;
+  }
+  return null;
+}
+
+export function getLargeWhitelistedModel(
+  owner: WorkspaceType
+): ModelConfigurationType | null {
```
Comment (JulesBelveze) [inline types/src/front/lib/assistant.ts:82]: Same comment as in `getSmallWhitelistedModel`

File: front/package-lock.json [hunk 2]
```diff
@@ -4,7 +4,6 @@
   "requires": true,
   "packages": {
     "": {
-      "name": "front",
```
Comment (JulesBelveze) [inline front/package-lock.json:7]: Do we want to get rid of this?

**Review by JulesBelveze — COMMENTED** [review]
LGTM, left a couple comments OOC. 
Seems to me that having a Record would make sure we have consistent providers everywhere and enforce to have one small/large model

File: types/src/front/lib/assistant.ts [hunk 3]
```diff
@@ -50,6 +51,50 @@ export function metaPromptForProvider(
...
+  owner: WorkspaceType,
+  providerId: ModelProviderIdType
+) {
+  const whiteListedProviders = owner.whiteListedProviders ?? MODEL_PROVIDER_IDS;
+  return whiteListedProviders.includes(providerId);
+}
+
+export function getSmallWhitelistedModel(
+  owner: WorkspaceType
+): ModelConfigurationType | null {
```
[the remark this answers is not in the document]
Comment (spolu) [reply types/src/front/lib/assistant.ts:64]: well the idea is to get whatever small or large model. If we have no provider we get null which will allow us to error at the call site?

File: front/package-lock.json (hunk 2 above)
Comment (spolu) [reply front/package-lock.json:7]: This one is a bit stochastic it comes an goes

**Review by JulesBelveze — APPROVED** [review]

**Pull request merged**